src/Lorentz/Contracts/TZBTC/V2/Impl.hs

```diff
@@ -0,0 +1,139 @@
+{- SPDX-FileCopyrightText: 2019 Bitcoin Suisse
+ -
+ - SPDX-License-Identifier: LicenseRef-MIT-BitcoinSuisse
+ -}
+{-# LANGUAGE RebindableSyntax #-}
+{-# OPTIONS_GHC -fno-warn-unused-do-bind #-}
+
+module Lorentz.Contracts.TZBTC.V2.Impl
+  ( Entrypoint
+  , StorageC
+  , ManagedLedger.approve
+  , ManagedLedger.getAllowance
+  , ManagedLedger.getBalance
+  , ManagedLedger.getTotalSupply
+  , ManagedLedger.transfer
+  , V1.acceptOwnership
+  , V1.addOperator
+  , burn
+  , V1.getOwner
+  , V1.getTotalBurned
+  , V1.getTotalMinted
+  , V1.getRedeemAddress
+  , V1.getTokenMetadata
+  , mint
```

Comment:
How are `mint` and `burn` here different from V1?

Reply:
`ledger` in storage is now accessible in a different way and we use `creditTo` from the recent ManagedLedger which compiles with this change in storage, so there might be semantically insignificant changes.

Reply:
Ah, ok, our `transfer` is from `ManagedLedger`, not from `V1`, I see.